torch/csrc/jit/passes/tensorexpr_fuser.cpp

```diff
@@ -466,6 +467,34 @@ class TensorExprFuser {
     return true;
   }
 
+  bool canFuseOnDevice(Value* v) {
+    auto type = v->type()->cast<TensorType>();
+    if (!type) {
+      return true;
+    }
+    auto device = type->device();
+    if (!device) {
```

Comment:
I know it's an existing logic, but I think we might want to be stricter in how we treat tensors with no device in the type, at least in the NNC fuser.